### ISSUE TITLE
Load on demand

### ISSUE THREAD

**Issue by warpech** [Starcounter/starcounter-layout-html-editor#7]
Let's the same technique as in https://github.com/Starcounter/starcounter-debug-aid (`Polymer.Base.importHref`) to load the custom element on demand.

**Comment by tomalec** [comment]
Shouldn't the importer decide when to import it, rather than importee? 
So not the `starcounter-layout-editor` should do that, but whoever is using it, for example Layout app.

Then as long as the `﻿<link rel="import" href="/sys/starcounter-layout-html-editor/starcounter-layout-html-editor.html" />` is added only after panel is opened it should be enough.

The solution in s-d-a looks pretty sketchy https://github.com/Starcounter/starcounter-debug-aid/pull/43/files#r94347727

We could also move import to the element markup, so it would be appended after the (panel) element is created, or change panel from CE to a partial.

**Comment by warpech** [comment]
Good idea actually! 

So the `CompositionEditor` app should just add lazy loading of this element?

**Comment by warpech** [comment]
I think that lazy loading through `this.importHref` is much cleaner than `lazy-imports.html`.

Lazy imports looks like a hack that might get poor support in future. It still does not have a Polymer 2.0 implementation, meaning that it is down on Polymer's priority list.

**Comment by tomalec** [comment]
> Lazy imports looks like a hack that might get poor support in future. It still does not have a Polymer 2.0 implementation, meaning that it is down on Polymer's priority list.

At Polymer Summit they strongly advertised it as a way to achieve PRPL & co. So I wouldn't sa yso.

Also as HTML Imports are now out of W3C track, and become more Polymer-specific, they could do tighten integration (=> performance benefits) between `import` an `lazy-import`.

Also as you can see at https://github.com/Polymer/lazy-imports/blob/master/lazy-imports.html#L52

`<link rel="lazy-import">` is not a custom element. `LazyImportsBehavior` is the way to get `importLazyGroup` method which is a promise aggregation sugar on top of `importHref` and existing `<lik rel="`s. So, it's just more declarative way to do exactly same thing, and execute more-or-less same code.

Anyway, I'm fine with imperative `importHref`, as long as it small, if we would need to aggregate few promises, we should use `importLazyGroup`

**Comment by tomalec** [comment]
>  It still does not have a Polymer 2.0 implementation, meaning that it is down on Polymer's priority list.

It currently is `PolymerBehavior` so I believe it's already 2.0 compatible

**Comment by warpech** [comment]
Maybe, but still incomplete (https://github.com/Polymer/lazy-imports/issues/2) and buggy (https://github.com/Polymer/lazy-imports/issues/5). There are few months old unmerged PRs with solutions. I would rather stick to what's built into Polymer.

**Comment by tomalec** [comment]
:+1:

**Comment by stramel** [comment]
[Polymer/lazy-imports#5](https://github.com/Polymer/lazy-imports/issues/5) has been resolved and I will be making sure the 2.0-preview is ready soon

**Comment by warpech** [comment]
Closing as obsoleted by another issue with a better description: https://github.com/Starcounter/Blending/issues/95
